apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/PartDEventTransformerV2Test.java

```diff
@@ -55,7 +55,7 @@ public PartDEvent generateClaim() throws FHIRException {
             .findFirst()
             .get();
 
-    claim.setLastUpdated(Instant.now());
+    claim.setLastUpdated(Optional.of(Instant.now()));
```

Comment:
How does this relate to changing the table and column names though?

Reply:
I guess I'm going to need Karl to explain the (current) 'lastUpdated' field. For example, here is a snippet of code from CarrierClaim.java:
`
@OneToMany(
      mappedBy = "parentClaim",
      orphanRemoval = true,
      fetch = FetchType.LAZY,
      cascade = CascadeType.ALL
  )
  @OrderBy("lineNumber ASC")
  private List<CarrierClaimLine> lines = new LinkedList<>();

  private Instant lastUpdated;

  public String getBeneficiaryId() {
    return beneficiaryId;
  }
`
And here is a snippet of DDL for 'CarrierClaims' table:
`
    "finalAction" character(1) COLLATE pg_catalog."default" NOT NULL,
    lastupdated timestamp with time zone,
    "claimCarrierControlNumber" character varying(23) COLLATE pg_catalog."default",
`
Now at first blush, we see that there is a table column that stores _lastupdated_; we also see that there is an entity column _lastUpdated_, but there is no _@Column_ tag in the entity bean so...hmm, why no explicit mapping? This works because in db parlance, (entity) _lastUpdated_ equals (db) _lastupdated_ because db columns are case agnostic unless specifically defined otherwise; i.e., ddl column name can use all upper-case, or all lower-case or even mixed case...they all work equally the same.

Personally, I don't like that kind of 'black magic'; to me it is best to explicitly declare what and how db data maps to the entity bean elements. So preferably _lastupdated_ becomes:
`
  @Column(
      name = "last_updated",
      nullable = true
  )
  private Instant lastUpdated;
`
The problem with that is that the entity bean, because of the 'nullable' property (which correctly maps to the db column definition) will result in getter/setter treating that data as optional:
`
  public Optional<Instant> getLastUpdated() {
    return Optional.ofNullable(lastUpdated);
  }

  public void setLastUpdated(Optional<Instant> lastUpdated) {
    this.lastUpdated = lastUpdated.orElse(null);
  }
`
Now, it just so happens that the current master RifLayoutProcessor.java has the following code and comment:
`
    // Setter method which is useful for testing, but not needed in the main modules
    final MethodSpec lastUpdatedSetter =
        MethodSpec.methodBuilder("setLastUpdated")
            .addModifiers(Modifier.PUBLIC)
            .addParameter(ParameterSpec.builder(Instant.class, "lastUpdated").build())
            .addStatement("this.lastUpdated = lastUpdated")
            .returns(TypeName.VOID)
            .build();
`
So the entity will have a setter that is effectively ONLY used for mocking tests data. So the fact that a test has a change to comply with the entity is at best interesting and has no impact on actual data handling within the app.

Now I get the notion that we are trying to minimize code diffs in PRs, but I also feel that code improvement should also be a goal of code reviews.

Reply:
ok, but this one line will translate into approx 40+ changes elsewhere; and to be quite honest I'm not sure how to do this.

Here is the situation:

- we changed _lastupdated_ to _last_updated_
- that requires us to do one of two things:
   - add the @Column to map the db column last_updated to an entity field lastUpdated;
   - don't add the @Column but make the entity property also called last_updated; so this will result in snake_case entity property because that is the only way to get the 'default' db column to entity field mapping. This also perpetuates the bad practice of an entity not representing what the db table contains (i.e., the entity does not specify the nullable even though the table defines the column as nullable).

Assuming we do the above, every claim (parent table) will be modified. Also, changes to RifLayoutProcessor.java will need to be done.

Reply:
hmm, ok, that doesn't sound too good either! but now I understand why this all stems from the column renaming so thank you for adding that bit of explanation and this seems more in scope now for this PR. I'm also interested to get thoughts from @karlmdavis

Reply:
Guess, I need to plan better; I've already made changes to the following files:
`	modified:   apps/bfd-model/bfd-model-codegen/src/main/java/gov/cms/bfd/model/codegen/RifLayoutsProcessor.java
	modified:   apps/bfd-model/bfd-model-rif/src/main/java/gov/cms/bfd/model/rif/RifRecordBase.java
	modified:   apps/bfd-pipeline/bfd-pipeline-ccw-rif/src/main/java/gov/cms/bfd/pipeline/ccw/rif/load/RifLoader.java
	modified:   apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/BeneficiaryTransformerV2Test.java
	modified:   apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/CarrierClaimTransformerV2Test.java
	modified:   apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/CoverageTransformerV2Test.java
	modified:   apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/DMEClaimTransformerV2Test.java
	modified:   apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/HHAClaimTransformerV2Test.java
	modified:   apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/HospiceClaimTransformerV2Test.java
	modified:   apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/InpatientClaimTransformerV2Test.java
	modified:   apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/OutpatientClaimTransformerV2Test.java
	modified:   apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/PartDEventTransformerV2Test.java
	modified:   apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/SamhsaMatcherR4FromClaimTransformerV2Test.java
	modified:   apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/stu3/providers/BeneficiaryTransformerTest.java
	modified:   apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/stu3/providers/CarrierClaimTransformerTest.java
	modified:   apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/stu3/providers/CoverageTransformerTest.java
	modified:   apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/stu3/providers/InpatientClaimTransformerTest.java
	modified:   apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/stu3/providers/SamhsaMatcherFromClaimTransformerTest.java`

Shuld I just continue doing changes or are we going with orig bfd1399? Thx.

Reply:
I'm of the mind to stick with the original if the alternative involves even more changes and risk